src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java

```diff
@@ -14,224 +14,199 @@
-      
+
       void invalidateAllWords();
       void invalidateMisspelledWords();
-      
+
       void releaseOnDismiss(HandlerRegistration handler);
    }
 
-   public SpellChecker(Context context)
+   public TypoSpellChecker(Context context)
    {
       RStudioGinjector.INSTANCE.injectMembers(this);
-      
+
       // save reference to context and read its dictionary
       context_ = context;
       contextDictionary_ = context_.readDictionary();
-          
-              
-      // subscribe to spelling service changes (these occur when when the
-      // spelling dictionaries are changed)
-      context_.releaseOnDismiss(spellingService_.addChangeHandler(
-                                                   new ChangeHandler() {
-         @Override
-         public void onChange(ChangeEvent event)
-         {
-            context_.invalidateAllWords();
-         }
-      }));
-      
+
       // subscribe to spelling prefs changes (invalidateAll on changes)
-      uiPrefs_.ignoreWordsInUppercase().addValueChangeHandler(
-                                                         prefChangedHandler_);
-      uiPrefs_.ignoreWordsWithNumbers().addValueChangeHandler(
-                                                         prefChangedHandler_);
-      
+      uiPrefs_.realTimeSpellChecking().addValueChangeHandler(prefChangedHandler_);
+      uiPrefs_.ignoreWordsInUppercase().addValueChangeHandler(prefChangedHandler_);
+      uiPrefs_.ignoreWordsWithNumbers().addValueChangeHandler(prefChangedHandler_);
+
       // subscribe to user dictionary changes
-      context_.releaseOnDismiss(userDictionary_.addListChangedHandler(
-                                                   new ListChangedHandler() {
-         @Override
-         public void onListChanged(ListChangedEvent event)
+      context_.releaseOnDismiss(userDictionary_.addListChangedHandler((ListChangedEvent event) ->
          {
             // detect whether this is the first delivery of the list
             // or if it is an update
             boolean isUpdate = userDictionaryWords_ != null;
-                 
+
             userDictionaryWords_ = event.getList();
-            
+
             updateIgnoredWordsIndex();
-            
+
             if (isUpdate)
-               context_.invalidateMisspelledWords(); 
+               context_.invalidateMisspelledWords();
          }
-      }));
+      ));
    }
-   
+
    @Inject
-   void intialize(SpellingService spellingService,
-                  WorkbenchListManager workbenchListManager,
-                  UIPrefs uiPrefs)
+   void initialize(SpellingService spellingService, WorkbenchListManager workbenchListManager, UIPrefs uiPrefs)
    {
-      spellingService_ = spellingService;
       userDictionary_ = workbenchListManager.getUserDictionaryList();
       uiPrefs_ = uiPrefs;
+
+      ExternalJavaScriptLoader.Callback loadTypoCallback = () -> typoNative = new TypoNative("en_US");
```

Comment:
Normally the initialize methods are used purely for dependency injection. Any reason not to just put this in the constructor (where the injector is probably called)?

Reply:
I believe I originally put it there and there was some race condition/out of order loading that happened. I'll try again though and see if it works.